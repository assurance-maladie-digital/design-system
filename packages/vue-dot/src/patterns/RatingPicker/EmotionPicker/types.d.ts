export interface EmotionItem {
	title: string;
	description: string;
	icon: string;
	color: string;
}
