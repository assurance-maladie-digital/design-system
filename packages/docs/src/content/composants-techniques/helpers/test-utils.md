---
title: testUtils
description: Le helper `testUtils` contient un ensemble de fonctions utilitaires pour les tests unitaires.
---

<doc-tabs>

<doc-tab-item label="Utilisation">

<doc-alert type="info">

Le helper `testUtils` exporte également toutes les fonctions du package [Vue Test Utils](https://vue-test-utils.vuejs.org/).

</doc-alert>

### applyGlobalConfig

La fonction `applyGlobalConfig` permet d’appliquer la configuration globale pour les tests, elle est utilisée dans le fichier `setup.ts`.

```ts
import { applyGlobalConfig } from '@cnamts/vue-dot/src/helpers/testUtils';

applyGlobalConfig();
```

### installGlobalPlugins

La fonction `installGlobalPlugins` permet d’installer les plugins [Vue Input Facade](https://ronaldjerez.github.io/vue-input-facade/), [Vue Meta](https://vue-meta.nuxtjs.org/), [Vuex](https://vuex.vuejs.org/), [Vuetify](https://v2.vuetifyjs.com/) et Vue Dot pour les tests.

```ts
import { createLocalVue, installGlobalPlugins } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installGlobalPlugins(localVue);
```

### installRouter

La fonction `installRouter` permet d’installer [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createLocalVue, installRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installRouter(localVue);
```

### createRouter

La fonction `createRouter` permet de créer une instance de [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const routes = [];

const router = createRouter({
	routes
});
```

### createStore

La fonction `createStore` permet de créer une instance de [Vuex](https://vuex.vuejs.org/) pour les tests.

```ts
import { createStore } from '@cnamts/vue-dot/src/helpers/testUtils';

const store = createStore({
	state: {}
});
```

### createVuetifyInstance

La fonction `createVuetifyInstance` permet de créer une instance de [Vuetify](https://v2.vuetifyjs.com/) pour les tests.

```ts
import { createVuetifyInstance } from '@cnamts/vue-dot/src/helpers/testUtils';

const vuetify = createVuetifyInstance();
```

### mockVFormRef

La fonction `mockVFormRef` permet de mocker la référence d’un composant [VForm](https://v2.vuetifyjs.com/en/components/forms/) pour les tests.

```ts
import { mockVFormRef } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	$refs: {
		form: mockVFormRef(true)
	}
};
```

### mockTranslations

La fonction `mockTranslations` permet de mocker les traductions pour les tests.

```ts
import { mockTranslations } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	...mockTranslations<string | string[]>({
		'path.to.title': 'Titre',
		'path.to.items': [
			'Élément 1',
			'Élément 2',
			'Élément 3'
		]
	})
};
```

</doc-tab-item>

<doc-tab-item label="API">
<doc-api name="helpers/test-utils"></doc-api>
</doc-tab-item>

</doc-tabs>
