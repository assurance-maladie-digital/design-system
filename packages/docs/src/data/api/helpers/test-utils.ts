import { Api } from '~/types';

export const api: Api = {
	applyGlobalConfig: {
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installGlobalPlugins: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installRouter: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	createRouter: {
		arguments: [
			{
				name: 'options',
				description: 'Router options',
				type: 'RouterOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'VueRouter',
			},
		],
	},
	createStore: {
		arguments: [
			{
				name: 'options',
				description: 'Store options',
				type: 'StoreOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'Store',
			},
		],
	},
	createVuetifyInstance: {
		arguments: [
			{
				name: 'preset',
				description: 'Vuetify preset',
				type: 'Partial<UserVuetifyPreset>',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'Vuetify',
			},
		],
	},
	mockVFormRef: {
		arguments: [
			{
				name: 'isValid',
				description: 'Is valid',
				type: 'boolean',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	mockTranslations: {
		arguments: [
			{
				name: 'translations',
				description: 'Translations',
				type: 'IndexedObject',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
};
