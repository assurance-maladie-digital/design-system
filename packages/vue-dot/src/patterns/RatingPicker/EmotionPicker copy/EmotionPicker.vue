<template>
	<div class="vd-emotion-picker">
		<div class="d-flex justify-center">
			<H6
				class="mb-4"
			>
				Cette page Ã  t'elle ete utile ?
			</H6>
		</div>
		<div class="d-flex justify-space-between">
			<div
				class="sad"
				@click="selectEmotion('sad')"
			>
				<div
					:class="{ active: isActive('sad') }"
					class="icon-button"
				>
					<div class="d-flex justify-center">
						<VIcon
							x-large
							class="mt-2"
							color="orange-darken-20"
						>
							{{ sadIcon }}
						</VIcon>
					</div>
					<div class="mt-1 d-flex justify-center">
						<span>
							Pas du tout
						</span>
					</div>
				</div>
			</div>
			<div
				class="neutral"
				@click="selectEmotion('neutral')"
			>
				<div
					:class="{ active: isActive('neutral') }"
					class="icon-button"
				>
					<div class="d-flex justify-center">
						<VIcon
							x-large
							class="mt-2"
							color="yellow-darken-20"
						>
							{{ neurtralIcon }}
						</VIcon>
					</div>
					<div class="mt-1 d-flex justify-center">
						<span>
							Moyen
						</span>
					</div>
				</div>
			</div>
			<div
				class="happy"
				@click="selectEmotion('happy')"
			>
				<div
					:class="{ active: isActive('happy') }"
					class="icon-button"
				>
					<div class="d-flex justify-center">
						<VIcon
							x-large
							class="mt-2"
							color="turquoise-darken-20"
						>
							{{ happyIcon }}
						</VIcon>
					</div>
					<div class="mt-1 d-flex justify-center">
						<span>
							Parfait !
						</span>
					</div>
				</div>
			</div>
		</div>
	</div>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component from 'vue-class-component';

	//icons
	import { mdiEmoticonSadOutline, mdiEmoticonNeutralOutline, mdiEmoticonHappyOutline } from '@mdi/js';

	@Component
	export default class FooterWrapper extends Vue {

		//icons
		sadIcon = mdiEmoticonSadOutline;
		neurtralIcon = mdiEmoticonNeutralOutline;
		happyIcon = mdiEmoticonHappyOutline;

		selectedEmotion = '';

		selectEmotion(emotion: string): void {
			this.selectedEmotion = emotion;
			this.$emit('update-emotion', emotion);
		}

		isActive(emotion: string): boolean {
			return this.selectedEmotion === emotion ? true : false;
		}
	}
</script>

<style lang="scss" scoped>
@import '@cnamts/design-tokens/dist/tokens';

h6 {
	font-size: 20px
}
.icon-button {
	height: 88px;
	width: 88px;
	border-radius: 8px;
}
.icon-button:hover {
	cursor: pointer;
}
.sad {
	.icon-button:hover {
		background-color: $vd-orange-lighten-90;
	}
	.active {
		border: 1px solid $vd-orange-darken-20;
		background-color: $vd-orange-lighten-90;
	}
}
.neutral {
	.icon-button:hover {
		background-color: $vd-yellow-lighten-90;
	}
	.active {
		border: 1px solid $vd-yellow-darken-20;
		background-color: $vd-yellow-lighten-90;
	}
}
.happy {
	.icon-button:hover {
		background-color: $vd-turquoise-lighten-90;
	}
	.active {
		border: 1px solid $vd-turquoise-darken-20;
		background-color: $vd-turquoise-lighten-90;
	}
}
</style>
