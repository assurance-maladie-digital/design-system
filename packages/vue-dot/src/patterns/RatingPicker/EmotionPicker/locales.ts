export const locales = {
	defaultEmotionLabels: [
		'Pas du tout',
		'Moyen',
		'Parfait !'
	]
};
