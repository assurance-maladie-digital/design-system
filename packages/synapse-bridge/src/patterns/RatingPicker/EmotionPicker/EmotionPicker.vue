<script lang="ts">
import { defineComponent } from "vue";
import type { PropType } from "vue";
import { RatingMixin } from "../RatingMixin";
import { locales } from "./locales";
import { propValidator } from "@/helpers/propValidator";
import {
	mdiEmoticonHappyOutline,
	mdiEmoticonSadOutline,
	mdiEmoticonNeutralOutline,
} from "@mdi/js";
export default defineComponent({
	mixins: [RatingMixin],
	emits: ["update:modelValue"],
	props: {
		length: {
			type: Number,
			default: 3,
			validator: (value: number) =>
				propValidator("length", ["2", "3"], value.toString()),
		},
		itemLabels: {
			type: Array as PropType<string[]>,
			default: () => locales.defaultEmotionLabels,
		},
	},
	data() {
		return {
			sadIcon: mdiEmoticonSadOutline,
			neutralIcon: mdiEmoticonNeutralOutline,
			happyIcon: mdiEmoticonHappyOutline
		}
	},
	computed: {
		btnSize(): string {
			const name = this.$vuetify.display.name;
			return name === 'xs' ? '70px' : '88px';
		},
		hasAnswered(): boolean {
			return this.modelValue !== -1;
		}
	},
	methods: {
		isActive(index: number): boolean {
			return index === this.modelValue - 1;
		},
		getIcon(index: number): string {
			if (index === 0) {
				return this.sadIcon;
			}
			if (index === 1 && this.length === 3) {
				return this.neutralIcon;
			}
			return this.happyIcon;
		},
		getColor(index: number): string {
			const colors = [
				'sad',
				'neutral',
				'happy'
			];
			if (this.length === 2) {
				const filteredColors = colors.filter(
					(item) => item !== 'neutral'
				);
				return filteredColors[index];
			}
			return colors[index];
		},
		getEmotionLabel(value: number): string {
			if (this.length === 2) {
				const filteredLabels = this.itemLabels.filter(
					(_, index) => index !== 1
				);
				return filteredLabels[value];
			}
			return this.itemLabels[value];
		}
	}
});
</script>

<template>
	<fieldset class="vd-emotion-picker">
		<legend class="text-h6 mb-6">
			<slot name="label">
				{{ label }}
			</slot>
		</legend>

		<VRating
			:model-value="modelValue"
			:length="length"
			:readonly="readonly"
			class="max-width-none mx-n1 mx-sm-n2"
		>
			<template #item="{ index }">
				<VBtn
					:disabled="readonly || hasAnswered"
					:aria-pressed="isActive(index).toString()"
					:class="[
						getColor(index),
						{ 'v-btn--active': isActive(index) },
					]"
					:min-height="btnSize"
					:min-width="btnSize"
					variant="text"
					class="rounded-lg px-1 px-sm-4 mx-1 mx-sm-2"
					@click="emitInputEvent(index + 1)"
				>
					<VIcon size="40" color="currentColor" class="pa-0">
						{{ getIcon(index) }}
					</VIcon>

					<span
						v-if="getEmotionLabel(index)"
						:class="{ 'text-secondary': !isActive(index) }"
						class="text-subtitle-2 mt-1"
					>
						{{ getEmotionLabel(index) }}
					</span>
				</VBtn>
			</template>
		</VRating>
	</fieldset>
</template>

<style lang="scss" scoped>
@import "@cnamts/design-tokens/dist/tokens";
.vd-emotion-picker {
	border: 0;
}
.v-rating .v-btn {
	transition: 0.2s;
	border: 1px solid transparent;
	opacity: 1;
	transition: 0.2s;
	background: transparent;
	:deep(.v-btn__content) {
		flex-direction: column;
	}
	.text-secondary {
		color: $vd-grey-lighten-20 !important;
	}
	&.sad {
		color: $vd-orange-darken-20 !important;
	}
	&.neutral {
		color: $vd-yellow-darken-20 !important;
	}
	&.happy {
		color: $vd-turquoise-darken-20 !important;
	}
	&--active.v-btn--disabled .v-icon {
		color: currentColor !important;
	}
	&:focus,
	&--active {
		border-color: currentColor !important;
	}
	&:hover,
	&:focus,
	&--active {
		:deep(.v-btn__overlay, .v-btn__underlay) {
			display: none;
		}
		&.sad {
			background: $vd-orange-lighten-90;
		}
		&.neutral {
			background: $vd-yellow-lighten-90;
		}
		&.happy {
			background: $vd-turquoise-lighten-90;
		}
	}
}
.v-theme--light.v-btn--disabled :deep(.v-icon) {
	color: $vd-grey-lighten-20 !important
}
</style>
