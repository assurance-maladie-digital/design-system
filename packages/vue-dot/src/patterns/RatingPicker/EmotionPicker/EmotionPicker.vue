<template>
	<fieldset class="vd-emotion-picker">
		<legend class="text-h6 mb-6">
			{{ label }}
		</legend>

		<VRating
			:value="value"
			:length="length"
			:readonly="readonly"
			class="max-width-none mx-n1 mx-sm-n2"
			@input="emitInputEvent"
		>
			<template #item="{ index, click }">
				<VBtn
					:disabled="readonly || hasAnswered"
					:aria-pressed="isActive(index).toString()"
					:class="[
						getColor(index),
						{ 'v-btn--active': isActive(index) }
					]"
					:min-height="btnSize"
					:min-width="btnSize"
					text
					class="rounded-lg px-1 px-sm-4 mx-1 mx-sm-2"
					@click="click"
				>
					<VIcon
						x-large
						color="currentColor"
						class="pa-0"
					>
						{{ getIcon(index) }}
					</VIcon>

					<span
						v-if="getEmotionLabel(index)"
						:class="{ 'text--secondary': !isActive(index) }"
						class="text-subtitle-2 mt-1"
					>
						{{ getEmotionLabel(index) }}
					</span>
				</VBtn>
			</template>
		</VRating>
	</fieldset>
</template>

<script lang="ts">
	import Vue, { PropType } from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { RatingMixin } from '../RatingMixin';
	import { locales } from './locales';

	import { propValidator } from '../../../helpers/propValidator';

	import {
		mdiEmoticonHappyOutline,
		mdiEmoticonSadOutline,
		mdiEmoticonNeutralOutline
	} from '@mdi/js';

	const Props = Vue.extend({
		props: {
			length: {
				type: Number,
				default: 3,
				validator: (value: number) => propValidator('length', ['2', '3'], value.toString())
			},
			itemLabels: {
				type: Array as PropType<string[]>,
				default: () => locales.defaultEmotionLabels
			}
		}
	});

	const MixinsDeclaration = mixins(Props, RatingMixin);

	@Component
	export default class EmotionPicker extends MixinsDeclaration {
		sadIcon = mdiEmoticonSadOutline;
		neutralIcon = mdiEmoticonNeutralOutline;
		happyIcon = mdiEmoticonHappyOutline;

		get btnSize(): string {
			return this.$vuetify.breakpoint.xs ? '70px' : '88px';
		}

		get hasAnswered(): boolean {
			return this.value !== -1;
		}

		isActive(index: number): boolean {
			return index === this.value - 1;
		}

		getIcon(index: number): string {
			if (index === 0) {
				return this.sadIcon;
			}

			if (index === 1 && this.length === 3) {
				return this.neutralIcon;
			}

			return this.happyIcon;
		}

		getColor(index: number): string {
			const colors = [
				'sad',
				'neutral',
				'happy'
			];

			if (this.length === 2) {
				const filteredColors = colors.filter((item) => item !== 'neutral');

				return filteredColors[index];
			}

			return colors[index];
		}

		getEmotionLabel(value: number): string {
			if (value === -1) {
				return '';
			}

			if (this.length === 2) {
				const filteredLabels = this.itemLabels.filter((_, index) => index !== 1);

				return filteredLabels[value];
			}

			return this.itemLabels[value];
		}
	}
</script>

<style lang="scss" scoped>
	@import '@cnamts/design-tokens/dist/tokens';

	.v-rating .v-btn {
		transition: 0.2s;
		border: 1px solid transparent;

		:deep(.v-btn__content) {
			flex-direction: column;
		}

		&.sad {
			color: $vd-orange-darken-20 !important;
		}

		&.neutral {
			color: $vd-yellow-darken-20 !important;
		}

		&.happy {
			color: $vd-turquoise-darken-20 !important;
		}

		&::before {
			opacity: 1;
			transition: 0.2s;
			background: transparent;
		}

		&--active.v-btn--disabled .v-icon {
			color: currentColor !important;
		}

		&:focus,
		&--active {
			border-color: currentColor !important;
		}

		&:hover,
		&:focus,
		&--active {
			&.sad::before {
				background: $vd-orange-lighten-90;
			}

			&.neutral::before {
				background: $vd-yellow-lighten-90;
			}

			&.happy::before {
				background: $vd-turquoise-lighten-90;
			}
		}
	}
</style>
