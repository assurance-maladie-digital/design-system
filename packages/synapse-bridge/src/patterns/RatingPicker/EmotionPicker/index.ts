import EmotionPicker from './EmotionPicker.vue';

export default EmotionPicker;
