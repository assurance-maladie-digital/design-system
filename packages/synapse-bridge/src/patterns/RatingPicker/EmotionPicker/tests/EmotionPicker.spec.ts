import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import { vuetify } from "@tests/unit/setup";

import EmotionPicker from "../";

describe("EmotionPicker", () => {
	it("renders correctly", () => {
		const wrapper = mount(EmotionPicker, {
			global: {
				plugins: [vuetify],
			},
			propsData: {
				label: "Pourriez-vous donner une note ?",
				itemLabels: ["Pas du tout", "Peut-être", "Oui super"],
			},
		});

		expect(wrapper.html()).toContain("Pourriez-vous donner une note ?");
		expect(wrapper.html()).toContain("Pas du tout");
		expect(wrapper.html()).toContain("Peut-être");
		expect(wrapper.html()).toContain("Oui super");
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders correctly with only 2 items", () => {
		const wrapper = mount(EmotionPicker, {
			global: {
				plugins: [vuetify],
			},
			propsData: {
				label: "Pourriez-vous donner une note ?",
				itemLabels: ["Pas du tout", "Not used", "Oui super"],
				length: 2,
			},
		});

		expect(wrapper.html()).toContain("Pourriez-vous donner une note ?");
		expect(wrapper.html()).toContain("Pas du tout");
		expect(wrapper.html()).toContain("Oui super");
		expect(wrapper.html()).not.toContain("Not used");
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders in mobile mode", async () => {
		const wrapper = mount(EmotionPicker, {
			global: {
				plugins: [vuetify],
			},
			propsData: {
				label: "Pourriez-vous donner une note ?",
				itemLabels: ["Pas du tout", "Peut-être", "Oui super"],
			},
		});

		wrapper.vm.$vuetify.display.name = 'xs';
		await wrapper.vm.$nextTick();

		expect(wrapper.html()).toContain("70px");
	});

	it("emit the right value when an item is clicked", async () => {
		const wrapper = mount(EmotionPicker, {
			global: {
				plugins: [vuetify],
			},
		});

		const items = wrapper.findAll("button");
		await items[0].trigger("click");
		expect(wrapper.emitted("update:modelValue")).toBeTruthy();
		expect(wrapper.emitted("update:modelValue")?.[0]?.[0]).toBe(1);
	});

	it("have no labels when no labels are provided", () => {
		const wrapper = mount(EmotionPicker, {
			global: {
				plugins: [vuetify],
			},
			propsData: {
				itemLabels: [],
			},
		});

		wrapper.findAll("button").forEach((button) => {
			expect(button.text()).toBe("");
		});
	});

	it("change the style of the buttons when an item is clicked", async () => {
		const wrapper = mount(EmotionPicker, {
			global: {
				plugins: [vuetify],
			},
		});

		const items = wrapper.findAll("button");
		const classBtn = items.map((item) => item.classes());
		await wrapper.setProps({ modelValue: 1 });
		expect(items[0].classes()).toContain("v-btn--active");
		const classBtnClicked = wrapper.findAll("button").map((item) => item.classes());

		for(let i = 0; i < classBtn.length; i++) {
			expect(classBtn[i]).not.toEqual(classBtnClicked[i]);
		}
	});
});
