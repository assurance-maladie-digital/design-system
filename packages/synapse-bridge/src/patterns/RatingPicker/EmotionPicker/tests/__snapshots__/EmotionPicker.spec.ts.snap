// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmotionPicker > renders correctly 1`] = `
<fieldset data-v-62a645b1="" class="vd-emotion-picker">
  <legend data-v-62a645b1="" class="text-h6 mb-6">Pourriez-vous donner une note ?</legend>
  <div data-v-62a645b1="" class="v-rating v-theme--light max-width-none mx-n1 mx-sm-n2"><label for="v-rating-0-0" class=""><span class="v-rating__hidden">Rating 0 of 3</span>
      <!---->
    </label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-0" type="radio" tabindex="-1" value="0">
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-1" class=""><span class="v-rating__hidden">Rating 1 of 3</span><button data-v-62a645b1="" type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text sad rounded-lg px-1 px-sm-4 mx-1 mx-sm-2" style="min-height: 88px; min-width: 88px;" aria-pressed="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i data-v-62a645b1="" class="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M15.5,8C16.3,8 17,8.7 17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M12,14C13.75,14 15.29,14.72 16.19,15.81L14.77,17.23C14.32,16.5 13.25,16 12,16C10.75,16 9.68,16.5 9.23,17.23L7.81,15.81C8.71,14.72 10.25,14 12,14Z mdi v-icon notranslate v-theme--light text-currentColor pa-0" style="font-size: 40px; height: 40px; width: 40px;" aria-hidden="true"></i><span data-v-62a645b1="" class="text-secondary text-subtitle-2 mt-1">Pas du tout</span></span>
            <!---->
            <!---->
          </button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-1" type="radio" tabindex="-1" value="1"></div>
      <!---->
    </div>
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-2" class=""><span class="v-rating__hidden">Rating 2 of 3</span><button data-v-62a645b1="" type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text neutral rounded-lg px-1 px-sm-4 mx-1 mx-sm-2" style="min-height: 88px; min-width: 88px;" aria-pressed="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i data-v-62a645b1="" class="M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M8.5,11A1.5,1.5 0 0,1 7,9.5A1.5,1.5 0 0,1 8.5,8A1.5,1.5 0 0,1 10,9.5A1.5,1.5 0 0,1 8.5,11M17,9.5A1.5,1.5 0 0,1 15.5,11A1.5,1.5 0 0,1 14,9.5A1.5,1.5 0 0,1 15.5,8A1.5,1.5 0 0,1 17,9.5M16,14V16H8V14H16Z mdi v-icon notranslate v-theme--light text-currentColor pa-0" style="font-size: 40px; height: 40px; width: 40px;" aria-hidden="true"></i><span data-v-62a645b1="" class="text-secondary text-subtitle-2 mt-1">Peut-Ãªtre</span></span>
            <!---->
            <!---->
          </button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-2" type="radio" tabindex="-1" value="2"></div>
      <!---->
    </div>
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-3" class=""><span class="v-rating__hidden">Rating 3 of 3</span><button data-v-62a645b1="" type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text happy rounded-lg px-1 px-sm-4 mx-1 mx-sm-2" style="min-height: 88px; min-width: 88px;" aria-pressed="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i data-v-62a645b1="" class="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8C16.3,8 17,8.7 17,9.5M12,17.23C10.25,17.23 8.71,16.5 7.81,15.42L9.23,14C9.68,14.72 10.75,15.23 12,15.23C13.25,15.23 14.32,14.72 14.77,14L16.19,15.42C15.29,16.5 13.75,17.23 12,17.23Z mdi v-icon notranslate v-theme--light text-currentColor pa-0" style="font-size: 40px; height: 40px; width: 40px;" aria-hidden="true"></i><span data-v-62a645b1="" class="text-secondary text-subtitle-2 mt-1">Oui super</span></span>
            <!---->
            <!---->
          </button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-3" type="radio" tabindex="-1" value="3"></div>
      <!---->
    </div>
  </div>
</fieldset>
`;

exports[`EmotionPicker > renders correctly with only 2 items 1`] = `
<fieldset data-v-62a645b1="" class="vd-emotion-picker">
  <legend data-v-62a645b1="" class="text-h6 mb-6">Pourriez-vous donner une note ?</legend>
  <div data-v-62a645b1="" class="v-rating v-theme--light max-width-none mx-n1 mx-sm-n2"><label for="v-rating-0-0" class=""><span class="v-rating__hidden">Rating 0 of 2</span>
      <!---->
    </label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-0" type="radio" tabindex="-1" value="0">
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-1" class=""><span class="v-rating__hidden">Rating 1 of 2</span><button data-v-62a645b1="" type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text sad rounded-lg px-1 px-sm-4 mx-1 mx-sm-2" style="min-height: 88px; min-width: 88px;" aria-pressed="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i data-v-62a645b1="" class="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M15.5,8C16.3,8 17,8.7 17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M12,14C13.75,14 15.29,14.72 16.19,15.81L14.77,17.23C14.32,16.5 13.25,16 12,16C10.75,16 9.68,16.5 9.23,17.23L7.81,15.81C8.71,14.72 10.25,14 12,14Z mdi v-icon notranslate v-theme--light text-currentColor pa-0" style="font-size: 40px; height: 40px; width: 40px;" aria-hidden="true"></i><span data-v-62a645b1="" class="text-secondary text-subtitle-2 mt-1">Pas du tout</span></span>
            <!---->
            <!---->
          </button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-1" type="radio" tabindex="-1" value="1"></div>
      <!---->
    </div>
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-2" class=""><span class="v-rating__hidden">Rating 2 of 2</span><button data-v-62a645b1="" type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text happy rounded-lg px-1 px-sm-4 mx-1 mx-sm-2" style="min-height: 88px; min-width: 88px;" aria-pressed="false"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i data-v-62a645b1="" class="M20,12A8,8 0 0,0 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M10,9.5C10,10.3 9.3,11 8.5,11C7.7,11 7,10.3 7,9.5C7,8.7 7.7,8 8.5,8C9.3,8 10,8.7 10,9.5M17,9.5C17,10.3 16.3,11 15.5,11C14.7,11 14,10.3 14,9.5C14,8.7 14.7,8 15.5,8C16.3,8 17,8.7 17,9.5M12,17.23C10.25,17.23 8.71,16.5 7.81,15.42L9.23,14C9.68,14.72 10.75,15.23 12,15.23C13.25,15.23 14.32,14.72 14.77,14L16.19,15.42C15.29,16.5 13.75,17.23 12,17.23Z mdi v-icon notranslate v-theme--light text-currentColor pa-0" style="font-size: 40px; height: 40px; width: 40px;" aria-hidden="true"></i><span data-v-62a645b1="" class="text-secondary text-subtitle-2 mt-1">Oui super</span></span>
            <!---->
            <!---->
          </button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-2" type="radio" tabindex="-1" value="2"></div>
      <!---->
    </div>
  </div>
</fieldset>
`;
